### DISCUSSION TITLE
Brush improvements

### DISCUSSION THREAD

**Discussion by Erevoid** [Orama-Interactive/Pixelorama#569] [Ideas]
Here are some ideas for improving the brushes:

Brush tilting: The ability to rotate the brushes

Middle click for project brush removal: With its current format, it's quite easy to accidentally erase the project brush you already have, so the ability of being able to remove the project brushes with **only** the middle click would solve that problem.

The ability to save your project brushes inside the file brushes from within the program: That could be potentially done by right clicking on the brush and a text appearing saying: save to file brushes. one step further would be the ability to add random brushes that way too. Similarly you could right click on a project brush and select: Save as file brush. A popup window can then open, enabling you to name your brush (if it's a random brush the % will be added automatically) and if you add another brush with the same name as a previous one, then it becomes a random brush with those two brushes (or more if it's done more than two times)

**Comment by OverloadedOrama** [comment]
More ideas:

- [ ] Load .gd files (GDScript files) as digital brushes. Digital brushes can be a new type of brush, that uses logic to handle drawing, scaling, and potentially other parameters that can be exposed as settings to the user. The circle brush is an example of a digital brush, which is currently coded inside Pixelorama's source. This idea is still under consideration.
- [ ] Add a refresh button that reloads brushes, like GIMP has.
- [ ] Make the UI more clear, so it's easier for the user to understand which type is each brush. A lot of people don't know about random brushes and they think of them as bugs at the beginning.
- [ ] Change how random brushes are being stored and loaded to solve #337 and to improve user experience.
- [ ] Easy way to save project brushes as file brushes. Could come hand in hand with the selection system improvements.
